Duplicati/Library/Main/Database/LocalListDatabase.cs

```diff
@@ -373,24 +404,37 @@ public IEnumerable<IFileset> Sets
                 get
                 {
                     var dict = new Dictionary<long, long>();
-                    for(var i = 0; i < m_filesets.Length; i++)
+                    for (var i = 0; i < m_filesets.Length; i++)
+                    {
                         dict[m_filesets[i].Key] = i;
+                    }
                     
-                    var summation = string.Format(@"SELECT ""A"".""FilesetID"" AS ""FilesetID"", COUNT(*) AS ""FileCount"", SUM(""C"".""Length"") AS ""FileSizes"" FROM ""FilesetEntry"" A, ""File"" B, ""Blockset"" C WHERE ""A"".""FileID"" = ""B"".""ID"" AND ""B"".""BlocksetID"" = ""C"".""ID"" AND ""A"".""FilesetID"" IN (SELECT DISTINCT ""FilesetID"" FROM ""{0}"") GROUP BY ""A"".""FilesetID"" ",m_tablename);
-                    
-                    using(var cmd = m_connection.CreateCommand())
-                    using (var rd = cmd.ExecuteReader(string.Format(@"SELECT DISTINCT ""A"".""FilesetID"", ""B"".""FileCount"", ""B"".""FileSizes"" FROM ""{0}"" A LEFT OUTER JOIN ( " + summation + @" ) B ON ""A"".""FilesetID"" = ""B"".""FilesetID"" ORDER BY ""A"".""Timestamp"" DESC ", m_tablename)))
-                        while(rd.Read())
+                    var summation =
+                        $@"SELECT ""A"".""FilesetID"" AS ""FilesetID"", COUNT(*) AS ""FileCount"", SUM(""C"".""Length"") AS ""FileSizes"" FROM ""FilesetEntry"" A, ""File"" B, ""Blockset"" C WHERE ""A"".""FileID"" = ""B"".""ID"" AND ""B"".""BlocksetID"" = ""C"".""ID"" AND ""A"".""FilesetID"" IN (SELECT DISTINCT ""FilesetID"" FROM ""{m_tablename}"") GROUP BY ""A"".""FilesetID"" ";
+
+                    using (var cmd = m_connection.CreateCommand())
+                    {
+                        using (var rd = cmd.ExecuteReader(
+                            $@"SELECT DISTINCT ""A"".""FilesetID"", ""A"".""IsFullBackup"", ""B"".""FileCount"", ""B"".""FileSizes"" FROM ""{m_tablename}"" A LEFT OUTER JOIN ( {summation} ) B ON ""A"".""FilesetID"" = ""B"".""FilesetID"" ORDER BY ""A"".""Timestamp"" DESC ")
+                        )
                         {
-                            var id = rd.GetInt64(0);
-                            var e = dict[id];
-                            
-                            var filecount = rd.ConvertValueToInt64(1, -1L);
-                            var filesizes = rd.ConvertValueToInt64(2, -1L);
+                            while (rd.Read())
+                            {
+                                var id = rd.GetInt64(0);
+                                if (!dict.ContainsKey(id))
+                                {
+                                    continue;
+                                }
+                                var isFullBackup = rd.GetInt32(1);
+                                var e = dict[id];
+
+                                var filecount = rd.ConvertValueToInt64(1, -1L);
+                                var filesizes = rd.ConvertValueToInt64(2, -1L);

```

Comment:
Since we added the `IsFullBackup` column to the result set, should we increment these column indices?

Reply:
Definitely. Good catch. Change is in.